src/core/column/sumprod.h

```diff
@@ -55,8 +55,14 @@ namespace dt {
           if (SUM) {            
             result = static_cast<T>(i1-i0) * value;
           } else {
-            for (size_t i = i0; i < i1; ++i) {
-              result *= value;
+            size_t power = i1-i0;
+            while (power) {
+              if (power % 2) {
+                result *= value;
+              }
+              if (!result) break;
+              value *= value;
+              power /= 2;
             }
```

Comment:
I guess this could be extracted as a separate function and placed into `models/utils.cc`. You can make it `inline` though.

Also, I'm not sure if `if (!result) break;` make it faster in average. I understand this is a shortcut for the case when `value` is `0`?

Reply:
was not aware of `models/utils.cc` . I'll check it out. 

You are right that the  `if(!result) break` is a shortuct for when value is 0; i'll remove it

Reply:
`model/utils` is just a good place to put a new function to, then we could re-use it if needed. 

The issue with `if (!result) break` is that it will be calculated every loop iteration for non-zero elements also. If you want to have a shortcut: you can check `value` outside of the loop. But I don't really know how much you will save.